bin/check-latest-npm.js

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env node
+
+/**
+ * External dependencies
+ */
+const { green, red, yellow } = require( 'chalk' );
+const { get } = require( 'https' );
+const { spawn } = require( 'child_process' );
+
+/**
+ * Returns a promise resolving with the version number of the latest available
+ * version of NPM.
+ *
+ * @return {Promise<string>} Promise resolving with latest NPM version.
+ */
+async function getLatestNPMVersion() {
+	return new Promise( ( resolve, reject ) => {
+		get(
+			'https://registry.npmjs.org/npm',
+			{
+				headers: {
+					// By passing a specialized `Accept` header, the registry
+					// will return an abbreviated form of the package data which
+					// includes enough detail to determine the latest version.
+					//
+					// See: https://github.com/npm/registry/blob/master/docs/responses/package-metadata.md
+					Accept: 'application/vnd.npm.install-v1+json',
+				},
+			},
+			async ( response ) => {
+				if ( response.statusCode !== 200 ) {
+					return reject(
+						new Error( 'Package data for NPM not found' )
+					);
+				}
+
+				let body = '';
+				for await ( const chunk of response ) {
+					body += chunk.toString();
+				}
+
+				let data;
+				try {
+					data = JSON.parse( body );
+				} catch {
+					return reject(
+						new Error(
+							'Package data for NPM returned invalid response body'
+						)
+					);
+				}
```

Comment:
Miscellaneous thought: The response of https://registry.npmjs.org/npm is quite large, even in its abbreviated form (using special `Accept` header). Since `dist-tags` is one of the very first properties of the JSON blob returned, I had some temptation to try to optimize this in some way to find the first occurrence of the `dist-tags` property and abandon processing immediately after. Thinking it may get ugly with regular expressions or a custom parser, I did find some semi-promising options, such as [this one](https://www.npmjs.com/package/bfj#how-do-i-selectively-parse-individual-items-from-a-json-stream). Ultimately, I think it's probably overkill at this point, but an interesting optimization all the same that could come in handy in other cases.